mergesort.py

```diff
@@ -0,0 +1,98 @@
+#! /usr/bin/env python
+import time
+from bst import Bst
+from list import makeworstlist
+
+
+
+def timed_func(func):
+    """Decorator for timing our insertion sort methods."""
+    def timed(*args, **kwargs):
+        start = time.time()
+        result = func(*args, **kwargs)
+        elapsed = time.time() - start
+        print "time expired: %s" % elapsed
+        return result
+    return timed
+
+
+def merge_sort(l):
+    """Sorts a list using a simple merge sort implemetnation"""
+    if len(l) <= 1:
+        return l
+    else:
+        left = l[:len(l)/2]
+        right = l[len(l)/2:]
```

Comment:
it's dangerous using true division.  Use floored specifically, since you clearly want an integer value no matter what.

```python
len(l)//2
```